src/java_bytecode/library/src/java/lang/String.java

```diff
@@ -0,0 +1,661 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.lang;
+
+import java.io.ObjectStreamField;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Locale;
+import org.cprover.CProver;
+
+public final class String
+    implements java.io.Serializable, Comparable<String>, CharSequence {
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
+    public String() {}
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
+    public String(String original) {}
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
+    public String(char value[]) {}
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
+    public String(char value[], int offset, int count) {}
+
+    public String(int[] codePoints, int offset, int count) {}
+
+    @Deprecated
+    public String(byte ascii[], int hibyte, int offset, int count) {}
+
+    @Deprecated
+    public String(byte ascii[], int hibyte) {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    private static void checkBounds(byte[] bytes, int offset, int length) {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    public String(byte bytes[], int offset, int length, String charsetName)
+            throws UnsupportedEncodingException {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    public String(byte bytes[], int offset, int length, Charset charset) {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    public String(byte bytes[], String charsetName)
+            throws UnsupportedEncodingException {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    public String(byte bytes[], Charset charset) {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    public String(byte bytes[], int offset, int length) {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    public String(byte bytes[]) {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    public String(StringBuffer buffer) {}
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
+    public String(StringBuilder builder) {}
+
+    // DIFFBLUE MODEL LIBRARY @TODO: implement this method internally in CBMC
+    String(char[] value, boolean share) {}
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
+    public int length() {
+        return CProver.nondetInt();
+    }
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
+    public boolean isEmpty() {
+        return CProver.nondetBoolean();
+    }
+
+    // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
```

Comment:
I don't think it is any more. Overridden method is now `CProverString.charAt`. There are other such cases.